File: client/Packages/IMCore/Sources/IMCore/Services/SessionService.swift
import Foundation

public final class SessionService: @unchecked Sendable {
    public static let shared = SessionService()
    private let api = APIClient.shared

    private init() {}

    public func listSessions() async throws -> [DeviceSession] {
        try await api.request("/api/v1/sessions")
    }

    public func deleteSession(_ sessionID: String) async throws {
        let _: [String: String] = try await api.request(
            "/api/v1/sessions/\(sessionID)",
            method: .delete
        )
    }
}

File: server/web/src/services/session-service.ts
import { api } from './api-client'
import type { Session } from '@/types/session'

export const sessionService = {
  list() {
    return api.request<Session[]>('/api/v1/sessions')
  },

  delete(sessionId: string) {
    return api.request<null>(`/api/v1/sessions/${sessionId}`, {
      method: 'DELETE',
    })
  },
}
